cmd/system/aws_cli.go

```diff
@@ -0,0 +1,329 @@
+			{{$ext = "pkg"}}
+			{{ end -}}
+
+			{{$base}}/{{$uri}}-{{$version}}.{{$ext}}
+		`,
+		}
+
+		dUrl, err := awsCliTool.GetURL(osVer, arch, version, !progress)
+		if err != nil {
+			return err
+		}
+		fmt.Printf("Downloading from: %s\n", dUrl)
+
+		outPath, err := get.DownloadFileP(dUrl, progress)
+		if err != nil {
+			return err
+		}
+		fmt.Printf("Downloaded to: %s\n", outPath)
+
+		f, err := os.OpenFile(outPath, os.O_RDONLY, 0644)
+		if err != nil {
+			return err
+		}
+		defer f.Close()
+
+		if workDir == "" {
+			cwd, err := os.Getwd()
+			if err != nil {
+				return err
+			}
+			workDir = cwd
+		}
+
+		fmt.Printf("Copying file to: %s\n", workDir)
+
+		filename := filepath.Base(outPath)
+		if _, err = get.CopyFileP(
+			outPath,
+			fmt.Sprintf("%s/%s", workDir, filename), readWriteExecuteEveryone,
+		); err != nil {
+			return err
+		}
+
+		isArchive, err := awsCliTool.IsArchive(true)
+		if err != nil {
+			return err
+		}
+
+		if isArchive {
+			unpackPath := fmt.Sprintf("%s/awscli", workDir)
+			fmt.Printf("Unpacking AWS CLI to: %s\n", unpackPath)
+
+			fInfo, err := f.Stat()
+			if err != nil {
+				return err
+			}
+			if err := archive.Unzip(f, fInfo.Size(), unpackPath, true); err != nil {
+				return err
+			}
+
+			workDir = unpackPath
+		}
+
+		if runInstaller {
+			if err := runBundledInstaller(osVer, workDir, filename, installPath); err != nil {
+				return err
+			}
+		} else {
+			tpl, err := installationInstructions(osVer, workDir, filename, installPath)
+			if err != nil {
+				return err
+			}
+			fmt.Printf("\n%s", tpl)
+		}
+
+		return nil
+	}
+
+	return command
+}
+
+func getAWSCLIVersion(owner, repo string) (string, error) {
+	url := fmt.Sprintf("https://api.github.com/repos/%s/%s/git/refs/tags", owner, repo)
+
+	client := http.Client{Timeout: time.Second * 10}
+	client.CheckRedirect = func(req *http.Request, via []*http.Request) error {
+		return http.ErrUseLastResponse
+	}
+
+	req, err := http.NewRequest(
+		http.MethodGet,
+		url,
+		nil,
+	)
+	if err != nil {
+		return "", err
+	}
+
+	var references []Reference
+	response, err := client.Do(req)
+	if err != nil {
+		return "", err
+	}
+
+	defer response.Body.Close()
+	body, err := io.ReadAll(response.Body)
+	if err != nil {
+		return "", err
+	}
+
+	if err := json.Unmarshal(body, &references); err != nil {
+		return "", err
+	}
+
+	tags := make([]*semver.Version, 0)
+	for _, reference := range references {
+		if reference.Object.Type == "tag" {
+			trimmed := strings.TrimPrefix(reference.Ref, "refs/tags/")
+			v, err := semver.NewVersion(trimmed)
+			if err != nil && errors.Is(err, semver.ErrInvalidSemVer) {
+				continue
+			}
+
+			tags = append(tags, v)
+		}
+	}
+
+	var comparer = func(a, b *semver.Version) int {
+		return a.Compare(b)
+	}
+
+	slices.SortFunc(tags, comparer)
+	latest := tags[len(tags)-1]
+
+	return latest.String(), nil
+}
+
+func runBundledInstaller(osVer string, workDir string, filename string, installPath string) error {
```

Comment:
I don't understand why this option is being included?

What format is the AWS CLI in?

Reply:
The CLI comes in different formats for different platforms, for instance OSx is a .pkg, Linux is a .zip, and similar for Windows. There isn't just a binary.